### PR TITLE
CreditDisplay.addCreditToNextFrame and CreditDisplay.addStaticCredit

### PR DESCRIPTION
**Pull request by ggetz**
This fixes https://github.com/CesiumGS/cesium/issues/6215 in preparation for adding credits to lightbox for use with the geocoder.

Given the confusion discussed in the issue, I deprecated existing functions in favor of `addCreditNextFrame` and `addStaticCredit`/`removeStaticCredit`, the latter of which now takes  `showOnScreen` into account.

This also
- adds a bit more documentation showing usage examples, and notes that `addCreditNextFrame` is mostly used in internal cases
- adds accessors Viewer.creditDisplay and CesiumWidget.creditDisplay for easier access

To test:

1.
`viewer.creditDisplay.addStaticCredit(new Cesium.Credit("My credit"));`
should show "My credit" in the lightbox

2.
`viewer.creditDisplay.addStaticCredit(new Cesium.Credit("My onscreen credit", true));`
should show "My onscreen credit" onscreen

`addCreditToNextFrame` is called internally, so

3. Confirm credits from imagery are added. For example the following is added to the lightbox for Bing Maps:
```
 © 2023 Microsoft Corporation
  Earthstar Geographics SIO
  <LOGO>
 ```

4. Load the ArcGIS MapServer example. There should be a default token credit warning on screen.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: packages/engine/Source/Scene/CreditDisplay.js [hunk 1]
```diff
@@ -450,24 +477,58 @@ CreditDisplay.prototype.addCredit = function (credit) {
...
+ * const credit = new Cesium.Credit('<a href="https://cesium.com/" target="_blank">Cesium</a>');
+ * viewer.creditDisplay.addStaticCredit(credit);
  */
-CreditDisplay.prototype.removeDefaultCredit = function (credit) {
+CreditDisplay.prototype.addStaticCredit = function (credit) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("credit", credit);
+  //>>includeEnd('debug');
+
+  const defaultCredits = this._defaultCredits;
```
Comment (jjhembd) [inline packages/engine/Source/Scene/CreditDisplay.js:520]: I like the new public-facing API naming. For the sake of future maintainers, can we make the internal naming consistent?  `this._defaultCredits` could be `this._staticCredits`

**Review by jjhembd — COMMENTED** [review]
Thanks @ggetz! The API change makes sense, and the tests work for me. I have just one comment about some internal naming

**Comment by ggetz** [comment]
Thanks @jjhembd! Updated!

**Comment by jjhembd** [comment]
Looks good, thanks @ggetz!

**Pull request merged**